linker/Mono.Linker/DirectoryAssemblyResolver.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Reflection;
+using System.Collections.Generic;
+using System.IO;
+using Mono.Collections.Generic;
+using Mono.Cecil;
+
```

Comment:
I agree about copying code. For now this gives us the flexibility to modify our assembly resolution for the scenarios we care about, as discussed with @jbevain. If cecil ends up supporting the resolution methods we need (see [306](https://github.com/jbevain/cecil/issues/306)) it should be easy enough to remove this.

I mostly took this directly from BaseAssemblyResolver.cs, but I must have changed a few of the conventions. I'll fix those - thanks!